### DISCUSSION TITLE
How to use mocks.SyncProducer.ExpectSendMessageAndFail to test multiple failures in SendMessages

### DISCUSSION THREAD

**Discussion by dharmjit** [IBM/sarama#2957] [Q&A]
Hi folks, I am unit testing my producer implementation, which uses `sarama.SyncProducer`, hence, I need to use the [github.com/Shopify/sarama/mocks](github.com/Shopify/sarama/mocks). This package provides the expect methods, but I am unsure how to use `ExpectSendMessageAndFail` for multiple failures in `SendMessages()`. I wanted to assert on all the `ProducerErrors`. Is there a way to test this?

**Comment by InboraStudio** [comment]
just call ```expectSendMessageAndFail``` so many times once for each mesage you expect to failed and then when you call ```SendMessages()``` fail each mesage in the same order thats it

**Reply by dharmjit** [reply]
That is the approach I am currently using, but having a Fluent API with Times() or something similar could have been useful as well.

**Reply by InboraStudio** [reply]
Problem is sarama not support that kinf of fluent api Its all manual

**Reply by InboraStudio** [reply]
you can loop it yourself
